classifiers/reference_quality/special_character_classifier/code_snippet_common.md

```diff
@@ -0,0 +1,41 @@
+```python
+import unicodedata
+from typing import List, Tuple
+
+DEFAULT_ALLOWED_RANGES = set(range(0x0020, 0x007F)).union( # Basic Latin
+    set(range(0x00A0, 0x00FF)), # Latin-1 Supplement
+    set(range(0x0100, 0x017F)),  # Latin Extended-A
+    set(range(0x0180, 0x024F)),  # Latin Extended-B
+    set(range(0x2000, 0x206F)),  # General Punctuation
+    set(range(0x20A0, 0x20CF)),  # Currency Symbols
+    set([ord("\t"), ord("\n"), ord("\r")])# common stop chars
+    )  
+
+
+def contains_special_characters(text: str, allowed_ranges: List[Tuple[int,int]] = None) -> str:
+    """
+    @param text: Text to detect special characters in
+    @param allowed_char_codes: Set of allowed char codes.
+    @return: True if text contains unusual characters, False otherwise.
+    """
+    
+    if allowed_ranges is None:
+        allowed_ranges = DEFAULT_ALLOWED_RANGES
+    
+    for char in text:
+        if ord(char) not in allowed_ranges and unicodedata.category(char) != "Zs":
+            return "true"
+    return "false"
```

Comment:
what is our approach in regards to same type in common / refinery?

Imo a function contain_special_characters is perfectly reasonable to be a bool (common) so the enduser can use at their own discression.
In refinery classification we need a label (which will be a variable/string).
In refinery generation a bool would be fine as well.

Reply:
Yes I agree with this. Common code now returns a boolean while refinery code returns a label!